### PR TITLE
Add ES_VERSION parameter to reindex.sh

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Add `ES_VERSION` parameter to `reindex.sh`.

**Why?**
`date` type needs to be converted to `date_nanos` for Elasticsearch 7 when migrating custom search attributes.
`double` type always needs to be converted to `scaled_float` when migrating custom search attributes.

**How did you test it?**
Run manually.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: schema/elasticsearch/visibility/versioned/v1/reindex.sh [hunk 1]
```diff
@@ -16,25 +23,43 @@ AUTO_CONFIRM="${AUTO_CONFIRM:-}"
...
+    if [ "${ES_VERSION}" != "v7" ]; then
+        DOC_TYPE="/_doc"
+    fi
+    CUSTOM_SEARCH_ATTRIBUTES_MAPPING=$(curl --silent --user "${ES_USER}":"${ES_PWD}" "${ES_ENDPOINT}/${ES_VIS_INDEX_V0}${DOC_TYPE}/_mapping" | jq --compact-output '.. | .Attr? | select(.!=null) | .properties | with_entries(select(.key == ($customSA[]))) | {properties:.}' --argjson customSA "${CUSTOM_SEARCH_ATTRIBUTES}")
+    if [ "${ES_VERSION}" == "v7" ]; then
+        # Replace "date" type with "date_nanos" only for Elasticsearch v7.
+        CUSTOM_SEARCH_ATTRIBUTES_MAPPING=$(jq '(.properties[].type | select(. == "date")) = "date_nanos"' <<< "${CUSTOM_SEARCH_ATTRIBUTES_MAPPING}")
+    fi
+    # Replace "double" type with "scaled_float".
+    CUSTOM_SEARCH_ATTRIBUTES_MAPPING=$(jq '(.properties[] | select(.type == "double")) = {"type":"scaled_float","scaling_factor":10000}' <<< "${CUSTOM_SEARCH_ATTRIBUTES_MAPPING}")
```
Comment (mmcshane) [inline schema/elasticsearch/visibility/versioned/v1/reindex.sh:43]: Is the lack of quoting on the right-hand side ok here?

File: schema/elasticsearch/visibility/versioned/v1/reindex.sh [hunk 2]
```diff
@@ -16,25 +23,43 @@ AUTO_CONFIRM="${AUTO_CONFIRM:-}"
 ES_ENDPOINT="${ES_SCHEME}://${ES_SERVER}:${ES_PORT}"
 DIR_NAME="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
 
+if ! curl --silent --user "${ES_USER}":"${ES_PWD}" "${ES_ENDPOINT}/_cluster/health" | jq --exit-status '.status=="green" | .'; then
```
Comment (mmcshane) [inline schema/elasticsearch/visibility/versioned/v1/reindex.sh:26]: minor but you probably want --fail here

**Review by mmcshane — APPROVED** [review]

File: schema/elasticsearch/visibility/versioned/v1/reindex.sh (hunk 2 above)
Comment (alexshtin) [reply schema/elasticsearch/visibility/versioned/v1/reindex.sh:26]: `--silent` should include it, no?

Comment (mmcshane) [reply schema/elasticsearch/visibility/versioned/v1/reindex.sh:26]: I don't think so. `curl --silent` will still given an exit code of zero for a 404. `curl --fail` will exit with 22.

**Review by underrun — APPROVED** [review]

**Pull request merged**